scrappers/itaka/scraper.py

```diff
@@ -0,0 +1,202 @@
+import time
+import requests
+import json
+import random
+
+from bs4 import BeautifulSoup
+from typing import List, Tuple
+
+from selenium.webdriver.common.action_chains import ActionChains
+from selenium import webdriver
+from selenium.webdriver.common.by import By
+from selenium.webdriver.support.ui import WebDriverWait
+from selenium.webdriver.support import expected_conditions as EC
+
+WEBSITE_URL: str = "https://www.itaka.pl/all-inclusive/"
+NO_OF_SCRAPED_OFFERS: int = 500
+
+
+class ItakaScraper:
+    def __init__(self, website_url: str, no_of_scraped_offers: int) -> None:
+        self.website_url: str = website_url
+        self.no_of_scraped_offers: int = no_of_scraped_offers
+        self.offer_urls: List[str] = self.get_itaka_urls()
+        self.scraped_data: List[dict] = self.generate_data()
+        self.save_json_as_file(filename="itaka_offers")
```

Comment:
nie wywoływałbym tego w konstruktorze, każde stworzenie instancji spowoduje utworzenie pliku a nie zawsze może być to pożądane